### DISCUSSION TITLE
Configuration Rendering :  Configuration Templates. Documentation example

### DISCUSSION THREAD

**Discussion by Domoninic** [netbox-community/netbox#17282] [Help Wanted!] [answered]
In the docs for [Configuration Rendering](https://github.com/netbox-community/netbox/blob/develop/docs/features/configuration-rendering.md) there is extends 'base.j2' and  include 'common/interface.j2'.
```
{% extends 'base.j2' %}

{% block content %}
    system {
        host-name {{ device.name }};
        domain-name example.com;
        time-zone UTC;
        authentication-order [ password radius ];
        ntp {
            {% for server in ntp_servers %}
                server {{ server }};
            {% endfor %}
        }
    }
    {% for interface in device.interfaces.all() %}
        {% include 'common/interface.j2' %}
    {% endfor %}
{% endblock %}
```
I get what these do with regard to jinja templating but in the context here where / how would base.j2 and interface.j2 be defined so they can be reference in Config Templates ?

**Comment by pl0xym0r** [comment, marked as answer]
In the same folder.
Path will depend on how you consume config template --> datasource or locally created
In case of datasource (git or local option, idk for S3) --> all the datafiles are stored in the same "root" folder, could be /opt/netbox/...  or the git repository itself --> just respect the relative path of your datafiles which is the same for the jinja files you write. 


In the doc example, it assumes a datasource is created, datafiles are there and the tree files is kind of :
/base.j2
/common/interface.j2
/exemple.j2
The config template that is mapped to a datafile mapped itself to a datasource will "understand" automatically all the files provided into the datasource, i.e you can import, extends, include these files into your j2
